Userland/Applications/FileManager/PropertiesWindowHashesTab.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2021, FrHun <frhun@t-online.de>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Optional.h>
+#include <AK/WeakPtr.h>
+#include <LibCore/File.h>
+#include <LibCrypto/Hash/HashManager.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/Event.h>
+#include <LibGUI/ImageWidget.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/TextBox.h>
+#include <LibThreading/BackgroundAction.h>
+
+class HashTab final : public GUI::Widget {
+    C_OBJECT(HashTab);
+
+public:
+    virtual ~HashTab() override;
+
+private:
+    class HashWidget;
+
+    const String m_path;
+    const bool m_calculate_on_show;
+
+    RefPtr<Gfx::Bitmap> m_icon_hash_incorrect;
+    RefPtr<Gfx::Bitmap> m_icon_hash_correct;
+    RefPtr<Gfx::Bitmap> m_icon_hash_load;
+    RefPtr<Gfx::Bitmap> m_icon_hash;
+    RefPtr<Gfx::Bitmap> m_icon_copy;
+
+    Optional<String> m_clipboard;
+
+    RefPtr<HashWidget> m_md5_widget;
+    RefPtr<HashWidget> m_sha1_widget;
+    RefPtr<HashWidget> m_sha256_widget;
+    RefPtr<HashWidget> m_sha512_widget;
+
+    HashTab(const String& path, const bool calculate_on_show);
+    HashTab(HashTab&&) = delete;
+    HashTab(HashTab&) = delete;
```

Comment:
My guess is that you're adding these to make it non-copyable and non-movable ?

In which case, you should use `AK_MAKE_NONCOPYABLE` and `AK_MAKE_NONMOVABLE` instead (see here for example : https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibCore/Object.h#L31)